.github/workflows/windows.yml

```diff
@@ -413,66 +413,137 @@ jobs:
+          tflags: ~1516 ~2301 ~2302 ~2303 ~2307
+          config: -DBUILD_SHARED_LIBS=OFF -DENABLE_DEBUG=ON -DCURL_ZSTD=ON -DUSE_NGHTTP2=ON -DENABLE_UNICODE=ON -DCURL_USE_LIBSSH2=ON -DCURL_USE_OPENSSL=ON -DUSE_OPENSSL_QUIC=ON
+        - name: no ssl
+          install: nghttp2 zstd zlib
+          arch: x64 
+          plat: windows
+          type: Debug
+          tflags: ~1516 ~2301 ~2302 ~2303 ~2307
+          config: -DENABLE_DEBUG=ON -DCURL_ZSTD=ON -DUSE_NGHTTP2=ON -DCURL_USE_SCHANNEL=OFF -DHTTP_ONLY=ON  -DENABLE_UNICODE=OFF
       fail-fast: false
     steps:
-      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
+      - uses: actions/checkout@v4
+
+      - name: Cache dependencies
+        uses: actions/cache/restore@v4
+        with:
+          path: "cache/${{ matrix.name }}"
+          key: ${{ matrix.plat }}-${{ matrix.name }}
+          restore-keys: ${{ matrix.plat }}-${{ matrix.name }}
+
+      - name: Init vcpkg
+        if: success()
+        shell: bash
+        run: |
+          git clone https://github.com/microsoft/vcpkg/
+          vcpkg/bootstrap-vcpkg.sh
+
+      - name: 'vcpkg build'
+        shell: bash
+        run: |
+          export VCPKG_ROOT=$PWD/vcpkg
+          export CURRENT_BINARY_CACHE="cache/${{ matrix.name }}"
+          mkdir -p "$CURRENT_BINARY_CACHE"
+          export VCPKG_BINARY_SOURCES="clear;files,$GITHUB_WORKSPACE/$CURRENT_BINARY_CACHE,readwrite;"          
+          vcpkg/vcpkg x-set-installed ${{ matrix.install }} --triplet=${{ matrix.arch }}-${{ matrix.plat }}
+          echo vcpkg/installed/vcpkg/:
+          ls vcpkg/installed/vcpkg/
+
+      - name: Save cache dependencies
+        id: cache-save
+        uses: actions/cache/save@v4
+        with:
+          path: "cache/${{ matrix.name }}"
+          key: ${{ matrix.plat }}-${{ matrix.name }}-${{ hashFiles('vcpkg/installed/vcpkg/status') }}
+
       - name: 'cmake configure'
         timeout-minutes: 5
         shell: bash
         run: |
-          archgen=${{ matrix.arch }}; [ "${archgen}" = 'x86' ] && archgen='Win32'
-          if [ '${{ matrix.plat }}' = 'uwp' ]; then
-            system='WindowsStore'
-            options='-DCMAKE_SYSTEM_VERSION=10.0'
-          else
-            system='Windows'
-          fi
-          [ '${{ matrix.type }}' = 'Debug' ] && options+=' -DCMAKE_RUNTIME_OUTPUT_DIRECTORY_DEBUG='
-          [ '${{ matrix.type }}' = 'Release' ] && options+=' -DCMAKE_RUNTIME_OUTPUT_DIRECTORY_RELEASE='
-          cmake -B bld ${options} \
-            "-DCMAKE_SYSTEM_NAME=${system}" \
-            -DCMAKE_TOOLCHAIN_FILE=C:/vcpkg/scripts/buildsystems/vcpkg.cmake \
-            "-DCMAKE_GENERATOR_PLATFORM=${archgen}" \
-            '-DVCPKG_TARGET_TRIPLET=${{ matrix.arch }}-${{ matrix.plat }}' \
-            -DCMAKE_VS_GLOBALS=TrackFileAccess=false \
-            '-DCMAKE_UNITY_BUILD=${{ matrix.unity }}' \
-            "-DCMAKE_C_FLAGS=${cflags}" \
-            '-DCMAKE_BUILD_TYPE=${{ matrix.type }}' \
```

Comment:
Can we keep the UWP logic?
Can we also keep the quotes, and `-DCMAKE_VS_GLOBALS=TrackFileAccess=false`.

Reply:
What does -DCMAKE_VS_GLOBALS=TrackFileAccess=false do?
I will restore the uwp logic.

Reply:
It disables internal administration done by MSBUILD, which makes the IDE (or subsequent builds?) work faster (as I understand).
This tracking work makes the CI job run slower and we don't use IDE, nor repeated builds, so it can be saved in this context.

Reply:
ok, Thank you for the explanations!